x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/PushFiltersToSource.java

```diff
@@ -252,8 +251,6 @@ static boolean canPushToSource(Expression exp, LucenePushdownPredicates lucenePu
                 && Expressions.foldable(cidrMatch.matches());
         } else if (exp instanceof SpatialRelatesFunction spatial) {
             return canPushSpatialFunctionToSource(spatial, lucenePushdownPredicates);
-        } else if (exp instanceof Term term) {
-            return term.field() instanceof FieldAttribute && DataType.isString(term.field().dataType());
```

Comment:
This was redundant indeed, I tried it quickly and it works fine (the verifier prevents these cases). 

nit: If you have a chance, it would be good to have a VerifierTest that checks that things like `WHERE term("foo", "bar")` return the right verification exception and don't result in a wrong pushdown.